pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -304,8 +320,29 @@ func revertToCutoverTimestamp(
 // TODO(adityamaru): Add ClearRange logic once we have introduced
 // synchronization between the flow tearing down and the job transitioning to a
 // failed/canceled state.
-func (s *streamIngestionResumer) OnFailOrCancel(_ context.Context, _ interface{}) error {
-	return nil
+func (s *streamIngestionResumer) OnFailOrCancel(ctx context.Context, execCtx interface{}) error {
+	jobExecCtx := execCtx.(sql.JobExecContext)
+	db := jobExecCtx.ExecCfg().DB
+	p := s.job.Payload()
+	details := s.job.Details().(jobspb.StreamIngestionDetails)
+
+	// Cancel the producer job even when the revert has error. The source job's
+	// protected timestamp is no longer needed as this ingestion job won't resume
+	// ingestion anymore.
+	defer cancelProducerJob(ctx,
+		streaming.StreamID(details.StreamID),
+		streamingccl.StreamAddress(details.StreamAddress))
+
+	// TODO(casper): deal with very large tenant.
+	// TODO(casper): ensure the tenant is offline while we revert it.
+	var b kv.Batch
+	b.AddRawRequest(&roachpb.ClearRangeRequest{
```

Comment:
Don't we want to to do this only if job was cancelled? And not e.g. because some error happen?